manager/allocator/networkallocator/portallocator.go

```diff
@@ -91,13 +91,17 @@ func reconcilePortConfigs(s *api.Service) []*api.PortConfig {
 		return s.Spec.Endpoint.Ports
 	}
 
-	allocatedPorts := make(map[api.PortConfig]*api.PortConfig)
+	allocatedPorts := make(map[api.PortConfig]map[uint32]*api.PortConfig)
 	for _, portState := range s.Endpoint.Ports {
 		if portState.PublishMode != api.PublishModeIngress {
 			continue
 		}
 
-		allocatedPorts[getPortConfigKey(portState)] = portState
+		portKey := getPortConfigKey(portState)
+		if _, ok := allocatedPorts[portKey]; !ok {
+			allocatedPorts[portKey] = make(map[uint32]*api.PortConfig)
```

Comment:
> Wondering; if `getPortConfigKey()` is intended to return the unique identifier for a port; should `PublishedPort` be just added to what `getPortConfigKey()` returns, instead of adding a `map` here?

This is a bit tricky. If `PublishedPort` was set to 0 when the service was created (to choose an ingress port dynamically), it will be set to 0 inside the spec and the actual allocated port inside `s.Endpoint`. These items must still be matched with each other, which is why `getPortConfigKey` intentionally excludes `PublishedPort`.

I think this is the right general approach.